topdown/graphql.go

```diff
@@ -339,29 +371,63 @@ func builtinGraphQLParseSchema(_ BuiltinContext, operands []*ast.Term, iter func
 }
 
 func builtinGraphQLIsValid(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
-	// Get the raw strings from each operand so that we can
-	// feed them to the GraphQL parser functions.
-	rawQuery, err := builtins.StringOperand(operands[0].Value, 1)
-	if err != nil {
-		return err
-	}
-	rawSchema, err := builtins.StringOperand(operands[1].Value, 1)
-	if err != nil {
-		return err
+	var queryDoc *gqlast.QueryDocument
+	var schemaDoc *gqlast.SchemaDocument
+	var schema *gqlast.Schema
+
+	switch x := operands[0].Value.(type) {
+	case ast.String:
+		// Get the raw strings from each operand so that we can
+		// feed them to the GraphQL parser functions.
+		rawQuery, err := builtins.StringOperand(x, 1)
+		if err != nil {
+			return err
+		}
```

Comment:
You've already checked that `operands[0]` is a string, so you could use `x` here -- without sing `builtins.StringOperand()` again. (Unless I'm missing something.)

Reply:
I've switched to using `x` where appropriate. You are correct, I already have ensured a particular type is present; I don't need to re-assert that. :slightly_smiling_face: